app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -258,12 +262,29 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
     )
   }
 
-  fun markHomeMenuCloseDrawer() {
+  fun markLastCheckedItemCloseDrawer(lastCheckedItemId: Int, isAdminSelected: Boolean) {
+    if (isAdminSelected) {
+      getFooterViewModel().isAdministratorControlsSelected.set(true)
+    } else if (lastCheckedItemId != -1) {
+      binding.fragmentDrawerNavView.menu.getItem(
+        when (lastCheckedItemId) {
+          NavigationDrawerItem.HOME.value -> 0
+          NavigationDrawerItem.OPTIONS.value -> 1
+          NavigationDrawerItem.HELP.value -> 2
+          NavigationDrawerItem.DOWNLOADS.value -> 3
+          NavigationDrawerItem.SWITCH_PROFILE.value -> 4
+          else -> 0
```

Comment:
giving else 0 means it will again give us the same issue or marking HOME as checked. Can we check that the execution will never go into the else part and if not we can remove the else part and use when with statements?

Reply:
Done